modules/dockermodelrunner/internal/sdk/client/pull.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+)
+
+// pullOptions contains options for pulling a model
+type pullOptions struct {
+	// Progress is an optional writer to show download progress
+	Progress ProgressWriter
+}
+
+// PullOption is a function that configures PullOptions
+type PullOption func(*pullOptions)
```

Comment:
question: could we ever have one which returned an error? If so do that now so we don't have to break the API later down the line.

Reply:
Oh indeed! we need that